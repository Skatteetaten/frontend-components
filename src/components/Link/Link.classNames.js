import { mergeStyleSets } from '@uifabric/merge-styles';
import { getTheme } from '@uifabric/styling';
import { MdIcons } from '../utils/icons';

export const getClassNames = props => {
  const { palette } = getTheme();
  const icon = "'" + MdIcons.icons[props.icon] + "'";

  return mergeStyleSets({
    iconLink: {
      display: 'block',
      position: 'relative',
      margin: '0 0 12px 26px',
      selectors: {
        '&::before': {
          fontWeight: 700,
          fontFamily: MdIcons.fontFace.fontFamily,
          display: 'inline-block',
          position: 'absolute',
          content: icon,
          color: palette.skeColor.blue,
          lineHeight: 1,
          top: '12px',
          left: '-26px',
          cursor: 'pointer'
        },
        '&:hover a': {
          borderBottom: '2px solid ' + palette.skeColor.blue
        }
      }
    },
    iconLinkA: {
      color: palette.skeColor.blue,
      textDecoration: 'none',
      fontWeight: 700
    }
  });
};
