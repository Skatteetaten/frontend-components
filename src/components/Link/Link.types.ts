import * as React from 'react';

export interface LinkProps
  extends React.AnchorHTMLAttributes<HTMLAnchorElement> {
  className?: string;
  /** Links are normally rendered as a-tags. This makes is possible to use a different markup. */
  renderContent?: (classNames: string) => JSX.Element;
  icon?: string;
  /** If the link should open in an new window (target=blank) */
  openInNew?: boolean;
  /** If the link is an invisible "skip to main content" link */
  skipLink?: boolean;
  path?: string;
  placement?: 'after' | 'before';
  text?: string;
  /** @ignore */
  linkGroup?: boolean;
}
