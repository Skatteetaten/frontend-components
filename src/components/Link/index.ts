export { Link as default } from './Link';
export * from './Link';
