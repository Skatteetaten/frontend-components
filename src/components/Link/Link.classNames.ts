import { mergeStyleSets } from '@uifabric/merge-styles';
import { getTheme } from '@uifabric/styling';
import { PaletteProps, FontSizes, FontWeights } from '../utils';
import { LinkProps } from './Link.types';

export const getClassNames = (props: LinkProps) => {
  const palette = getTheme().palette as PaletteProps;
  const { linkGroup } = props;

  return mergeStyleSets({
    icon: {
      color: palette.skeColor.blue,
      fontSize: linkGroup ? FontSizes.medium : FontSizes.large,
      fontWeight: linkGroup ? FontWeights.bold : 'inherit',
      verticalAlign: linkGroup ? 'bottom' : 'middle',
      paddingLeft: linkGroup ? '0px' : '4px',
      paddingRight: linkGroup ? '10px' : '4px',
      borderBottom: 'none',
    },
    linkContainer: {
      margin: '0px',
      selectors: {
        ':hover>i': {
          color: linkGroup ? palette.skeColor.darkBlue : '',
          paddingLeft: '4px',
          paddingRight: linkGroup ? '6px' : '4px',
          transition: linkGroup ? '0.1s' : '',
        },
        ':hover>a': {
          backgroundColor: palette.skeColor.lightBlue,
        },
      },
    },
    skipLink: {
      left: '-999px',
      position: 'absolute',
      top: 'auto',
      width: '1px',
      height: '1px',
      overflow: 'hidden',
      selectors: {
        '&:focus, &:active': {
          color: palette.skeColor.blue,
          backgroundColor: '#fff',
          position: 'relative',
          left: 'auto',
          top: 'auto',
          width: '100px',
          height: 'auto',
          overflow: 'auto',
          padding: '5px',
          border: `4px solid ${palette.skeColor.pink}`,
          textAlign: 'center',
        },
      },
    },
    iconLink: {
      color: palette.skeColor.blue,
      textDecoration: 'none',
      fontWeight: linkGroup ? FontWeights.bold : FontWeights.medium,
      borderBottom: `1px solid ${palette.skeColor.blue}`,
      selectors: {
        '&:hover': {
          color: palette.skeColor.darkBlue,
          borderBottom: '2px solid ' + palette.skeColor.darkBlue,
          backgroundColor: palette.skeColor.lightBlue,
        },
        '&:focus': {
          color: palette.skeColor.darkBlue,
          borderBottom: '2px solid ' + palette.skeColor.darkBlue,
          backgroundColor: palette.skeColor.lightBlue,
          outline: 'none',
        },
      },
    },
  });
};
