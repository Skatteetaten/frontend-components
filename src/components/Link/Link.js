import * as React from 'react';
import classnames from 'classnames';
import { getClassNames } from './Link.classNames';
import Step from '../StepList/Step';
import PropTypes from 'prop-types';

Step.propTypes = {
  /** Mulighet for å legge inn egen klasse for å overstyre stylingen */
  className: PropTypes.string,
  /** Ikon **/
  icon: PropTypes.string,
  /** Url **/
  path: PropTypes.string,
  /** Lenketekst **/
  text: PropTypes.string
};

const Link = props => {
  const [styles] = React.useState(getClassNames(props));
  return (
    <p className={classnames(styles.iconLink, props.className)}>
      <a href={props.path} className={classnames(styles.iconLinkA)}>
        {props.text}
      </a>
    </p>
  );
};

export default Link;
