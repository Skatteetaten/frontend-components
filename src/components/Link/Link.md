** Link med ikon **

```js
import Link from '@skatteetaten/frontend-components/Link';

<Link path={'#'} text={'Last ned dokumentene'} icon={'Download'} />;
```
