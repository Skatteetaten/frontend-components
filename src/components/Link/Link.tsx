import * as React from 'react';
import classnames from 'classnames';
import Icon from '../Icon';
import { getClassNames } from './Link.classNames';

export interface LinkProps
  extends React.AnchorHTMLAttributes<HTMLAnchorElement> {
  className?: string;
  /** Som standard rendres lenkene som a-elementer. Dette gir mulighet for å overstyre implementasjonen. */
  renderContent?: (classNames: string) => JSX.Element;
  icon?: string;
  /** Om lenken skal åpnes i nytt vindu (target=blank) */
  openInNew?: boolean;
  path?: string;
  placement?: 'after' | 'before';
  text?: string;
}

const Link: React.FC<LinkProps> = props => {
  const {
    className,
    placement,
    icon,
    path,
    text,
    openInNew,
    renderContent,
    ...htmlAttributes
  } = props;
  const styles = getClassNames();
  return (
    <span className={classnames(styles.linkContainer, props.className)}>
      {props.placement === 'before' && props.icon && (
        <Icon
          iconName={props.icon}
          className={styles.icon}
          role="presentation"
        />
      )}
      {renderContent ? (
        renderContent(classnames(styles.iconLink))
      ) : openInNew ? (
        <a
          href={props.path}
          className={classnames(styles.iconLink)}
          target="_blank"
          rel="noopener noreferrer"
          {...htmlAttributes}
        >
          {props.text}
        </a>
      ) : (
        <a
          href={props.path}
          className={classnames(styles.iconLink)}
          {...htmlAttributes}
        >
          {props.text}
        </a>
      )}

      {props.placement === 'after' && props.icon && (
        <Icon
          iconName={props.icon}
          className={styles.icon}
          role="presentation"
        />
      )}
    </span>
  );
};
export default Link;
