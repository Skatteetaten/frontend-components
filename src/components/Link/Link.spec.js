import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Link from './Link';

function oppsettMount() {
  return mount(
    <Link path={'#'} text={'Last ned dokumentene'} icon={'Download'} />
  );
}

describe('Link komponent', () => {
  it('matcher snapshot', () => {
    const wrapper = oppsettMount();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
