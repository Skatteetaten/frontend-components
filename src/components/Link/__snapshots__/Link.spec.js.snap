// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link komponent matcher snapshot 1`] = `
<Link
  icon="Download"
  path="#"
  text="Last ned dokumentene"
>
  <p
    className="iconLink-36"
  >
    <a
      className="iconLinkA-37"
      href="#"
    >
      Last ned dokumentene
    </a>
  </p>
</Link>
`;
